submission_worker.py

```diff
@@ -0,0 +1,300 @@
+from __future__ import absolute_import
+import django
+import importlib
+import json
+import os
+import pika
+import requests
+import signal
+import socket
+import time
+import yaml
+import zipfile
+
+from os.path import join
+from subprocess import Popen, call
+
+from django.conf import settings
+from django.utils import timezone
+from django.core.files.base import ContentFile
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'settings.dev')
+django.setup()
+
+from challenges.models import Challenge     # noqa
+from jobs.models import Submission          # noqa
+
+
+BASE_CHALLENGE_DATA_DIR = join(os.getcwd(), 'challenge_data')
+BASE_SUBMISSION_DATA_DIR = join(os.getcwd(), 'submission_files')
+CHALLENGE_DATA_DIR = join(BASE_CHALLENGE_DATA_DIR, '{challenge_id}')
+PHASE_DATA_DIR = join(os.getcwd(), 'challenge_data', '{challenge_id}', 'phase_data', '{phase_id}')
+SUBMISSION_DATA_DIR = join(BASE_SUBMISSION_DATA_DIR, '{submission_id}')
+
+django.db.close_old_connections()
+
+
+class ExecutionTimeLimitExceeded(Exception):
+    pass
+
+
+def alarm_handler(signum, frame):
+    raise ExecutionTimeLimitExceeded
+
+
+def download_and_extract_file(url, download_location, extract_location):
+    '''
+        * Function to extract downlaod a zip file, extract it and then removes the zip file.
+        * `download_location` should include name of file as well.
+    '''
+    try:
+        response = requests.get(url)
+    except Exception as e:
+        print 'Failed to fetch file from {}, error {}'.format(url, e)
+        # return from here or not?
+        response = None
+
+    if response and response.status_code == 200:
+        with open(download_location, 'w') as f:
+            f.write(response.content)
+        # extract zip file
+        zip_ref = zipfile.ZipFile(download_location, 'r')
```

Comment:
Is there a possibility that the content of the response is not a zip file? Are we putting enough checks in place to ensure that this is always a zip file?